tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/query/StringQueryExpressionTest.java

```diff
@@ -48,6 +48,11 @@ public void testSimpleQueryExprForStringResult() {
         Assert.assertEquals(returnValues.toString(), "Alex Ranjan John ");
     }
 
+    @Test(description = "Test simple query expression with string result")
```

Comment:
Lets add a negative test for ambiguous types also.

 ```bal
  string:Char[]|string strValue = check from Book _ in bookStream
        select chr;
```

Reply:
We can add it to `xml|xml[]` case too

Reply:
Added tests for ambiguous situations in both `string`,`xml`.
But those tests will uncomment after fixing #40012